lib/core.js

```diff
@@ -295,18 +295,26 @@ declare class JSON {
 }
 
 /* Iterators */
-interface IteratorResult<T> {
+interface IteratorResult<Y,R> {
```

Comment:
That said, #577 hasn't changed yet, and interfaces as type unions isn't possible either, so I'll commit this change and you can take it at your discretion.